SlackBotPrototype/SlackBotPrototype/Program.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Linq;
+using System.Threading;
+using API;
+using SlackAPI;
+using SlackAPI.WebSocketMessages;
+
+namespace SlackBotPrototype
+{
+	internal static class Program
+	{
+		private static SlackSocketClient _client;
+		private static IMessageToCommandConverter _messageToCommandConverter;
+		private static IWeatherProvider _weatherProvider;
+
+		private static void Main(string[] args)
+		{
+			var clientReady = new ManualResetEventSlim(false);
+			var slackToken = ConfigConstants.SlackApiSecret;
+			var darkSkyToken = ConfigConstants.DarkSkyApiSecret;
+			
+			if (string.IsNullOrWhiteSpace(slackToken))
+			{
+				// SHORTCUT: Normally would use logging library instead so that we can have a robust, and standardized logs
+				Console.WriteLine("Slack token not found in environment var name = `{SLACK_API_TOKEN}`");
+				return;
+			}
+
+			if (string.IsNullOrWhiteSpace(darkSkyToken))
+			{
+				Console.WriteLine("Dark sky API token not available var name = `{DARK_SKY_TOKEN}`");
+				return;
+			}
+
+			// SHORTCUT: We'd normally wrap the slack client library so that we can
+			// switch implementations or libraries with less fuss
+			_client = new SlackSocketClient(slackToken);
+			_weatherProvider = new DarkSkyWeatherProvider(darkSkyToken, new RestClientHttpProvider(ConfigConstants.DarkSkyBaseUrl));
+			_messageToCommandConverter = new MessageToCommandConverter(_weatherProvider, new SqLitePersistence());
+
+			_client.Connect((connected) => {

```

Comment:
Up to this point, I see configuration and initialization of the connection to `slack`. Makes sense. The closure (as the comment states) is called once the connection is established.

What happens after this? We get a user list. We add a message handler. There is setting and waiting. Give me a brief overview of how we're interacting with the library after we've established the connection.